Library/Homebrew/extend/pathname.rb

```diff
@@ -379,9 +379,12 @@ def write_env_script(target, args, env = nil)
     env_export = +""
     env.each { |key, value| env_export << "#{key}=\"#{value}\" " }
     dirname.mkpath
+
+    # We pass -a here to "trick" the target program into not seeing
+    # the env script at all.
     write <<~SH
       #!/bin/bash
-      #{env_export}exec "#{target}" #{args} "$@"
+      #{env_export}exec -a "#{self}" #{target}" #{args} "$@"
```

Comment:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Regardless of whether this part overcomes maintainer objections or not&thinsp;—&thinsp;though, given discussion, I suspect the latter will end up being the case&thinsp;—, you seem to've lost a double quote while making your changes here:  

```suggestion
      #{env_export}exec -a "#{self}" "#{target}" #{args} "$@"
```